### DISCUSSION TITLE
Bypass Repeated Logins & Captchas by Saving Cookies using Pickle

### DISCUSSION THREAD

**Discussion by aapatre** [aapatre/Automatic-Udemy-Course-Enroller-GET-PAID-UDEMY-COURSES-for-FREE#192] [Ideas]
Instead of logging in on each run, we can save the cookies using Python's Pickle module. So the login procedure will only be triggered on the first run and we can save time on other routine runs of the script.

Here's the info:
https://stackoverflow.com/questions/15058462/how-to-save-and-load-cookies-using-python-selenium-webdriver

![image](https://user-images.githubusercontent.com/20199688/103408000-c18ee380-4b86-11eb-8759-762757b148bf.png)

**Comment by cullzie** [comment]
Ye this is a good idea. We could get the cookie via REST which would avoid maintainence on the UI login stuff.
Would need to have a try/except in case the cookie expires though. I 'd imagine the login cookie has an expiry

**Reply by aapatre** [reply]
I see, the expiry thing can be an issue

**Comment by cullzie** [comment]
This is what is currently on alpha branch @aapatre
